.github/workflows/main.yml

```diff
@@ -1,6 +1,10 @@
 name: Run CI
 
-on: push
+on:
```

Comment:
I would just simplify this to:

```
on: [push, pull_request]
```

That way it just runs on every branch and every PR.

Reply:
I do not think we need to run on every branch. Just master and PR. Do you see any advantage in running in all branches? You would create a PR anyway?

Reply:
The only advantage being if you have a topic branch that isn't final and ready for a PR, but I guess that's a pretty rare case. Feel free to leave it as is. My main motivation was "why *not* let it run on all branches" :-)